### PR TITLE
Add freeimage 3.18.0

### PR DESCRIPTION
**Pull request by intelligide**
Specify library name and version:  **freeimage/3.18.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/freeimage/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        self.copy("license-fi.txt", dst="licenses", src=self._source_subfolder)
+        self.copy("license-gplv3.txt", dst="licenses", src=self._source_subfolder)
+        self.copy("license-gplv2.txt", dst="licenses", src=self._source_subfolder)
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
```
Comment (madebr) [inline recipes/freeimage/all/conanfile.py:92]:
```suggestion
        self.cpp_info.libs = ["freeimageplu", "freeimage"]
```

File: recipes/freeimage/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+        self.cpp_info.names["pkg_config"] = "freeimage"
+        self.cpp_info.names["cmake_find_package"] = "FreeImage"
+        self.cpp_info.names["cmake_find_package_multi"] = "FreeImage"
```
Comment (madebr) [inline recipes/freeimage/all/conanfile.py:96]: These names are not *official*, aren't they?
```suggestion
```

File: recipes/freeimage/all/patches/001_disable_TIFF.patch [hunk 3]
```diff
@@ -0,0 +1,22 @@
+--- Source/FreeImage/Plugin.cpp
++++ Source/FreeImage/Plugin.cpp
+@@ -254,7 +254,9 @@
+ 			s_plugins->AddNode(InitPNM, NULL, "PPMRAW", "Portable Pixelmap (RAW)", "ppm", "^P6");
+ 			s_plugins->AddNode(InitRAS);
+ 			s_plugins->AddNode(InitTARGA);
++#if 0
+ 			s_plugins->AddNode(InitTIFF);
```
Comment (madebr) [inline recipes/freeimage/all/patches/001_disable_TIFF.patch:8]: Isn't our libtiff recipe working?

File: recipes/freeimage/all/patches/001_disable_TIFF.patch [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
++#endif
+ 			s_plugins->AddNode(InitWBMP);
+ 			s_plugins->AddNode(InitPSD);
+ 			s_plugins->AddNode(InitCUT);
+@@ -263,7 +265,9 @@
+ 			s_plugins->AddNode(InitDDS);
+ 	        s_plugins->AddNode(InitGIF);
+ 	        s_plugins->AddNode(InitHDR);
++#if 0
+ 			s_plugins->AddNode(InitG3);
```
Comment (madebr) [inline recipes/freeimage/all/patches/001_disable_TIFF.patch:18]: What library are we missing?
Can you open an issue?

File: recipes/freeimage/all/patches/003_link_conan.patch [hunk 5]
```diff
@@ -0,0 +1,155 @@
+--- Source/FreeImage/J2KHelper.cpp
```
Comment (madebr) [inline recipes/freeimage/all/patches/003_link_conan.patch:1]: If it's not too much trouble, can you upstream this patch?
I think the biggest trouble will be modifying all makefiles to add the include paths.

File: recipes/freeimage/all/patches/001_disable_TIFF.patch (hunk 3 above)
Comment (intelligide) [reply recipes/freeimage/all/patches/001_disable_TIFF.patch:8]: FreeImage need a private copy of libtiff.

Comment (madebr) [reply recipes/freeimage/all/patches/001_disable_TIFF.patch:8]: Too many modifications? Dammit.

Comment (intelligide) [reply recipes/freeimage/all/patches/001_disable_TIFF.patch:8]: It's painful. Look at [vcpkg port](https://github.com/microsoft/vcpkg/tree/master/ports/freeimage), It's really annoying

Comment (madebr) [reply recipes/freeimage/all/patches/001_disable_TIFF.patch:8]: See https://sourceforge.net/p/freeimage/discussion/36111/thread/c326f1be79/
So it's unlikely that something will be done to fix this.

Comment (intelligide) [reply recipes/freeimage/all/patches/001_disable_TIFF.patch:8]: I don't really know how to fix it:
- Disabling some features and patching sources
or
- Using embedded deps sources

Both have huge drawbacks.

Comment (madebr) [reply recipes/freeimage/all/patches/001_disable_TIFF.patch:8]: Okay, if you're saying that it's a lot of work to make it truly use external dependencies.

File: recipes/freeimage/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure(build_dir=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        # tools.rmdir(os.path.join(self._source_subfolder, "Source", "LibJPEG"))
```
Comment (madebr) [inline recipes/freeimage/all/conanfile.py:69]: Can libjpeg be removed? You're adding libjpeg to the requires list.
Same question for the Zlib directory.

File: recipes/freeimage/all/conanfile.py (hunk 2 above)
Comment (intelligide) [reply recipes/freeimage/all/conanfile.py:96]: That's right. I just use the most popular pattern. (Ex: https://github.com/OGRECave/ogre/blob/master/CMake/Packages/FindFreeImage.cmake)

**Comment by intelligide** [comment]
#3027

File: recipes/freeimage/all/patches/001_disable_TIFF.patch (hunk 3 above)
Comment (SpaceIm) [reply recipes/freeimage/all/patches/001_disable_TIFF.patch:8]: I think that big patches to unvendor all these deps -like in the vcpkg port- are fine. The last version of freeimage is 2.5 years old.

**Pull request closed without merging**